redfish-core/lib/pcie.hpp

```diff
@@ -155,12 +208,24 @@ inline void requestRoutesSystemPCIeDevice(App& app)
                          "/redfish/v1/Systems/system/PCIeDevices/" + device +
                              "/PCIeFunctions"}};
                 };
-                std::string escapedPath = std::string(pciePath) + "/" + device;
-                dbus::utility::escapePathForDbus(escapedPath);
-                crow::connections::systemBus->async_method_call(
-                    std::move(getPCIeDeviceCallback), pcieService, escapedPath,
-                    "org.freedesktop.DBus.Properties", "GetAll",
-                    pcieDeviceInterface);
+
+                const auto& it = mapOfPcieDevicePaths.find(device);
```

Comment:
binding an r-value to a reference doesn't make much sense here.  I'm assuming you got the "can't bind non-const reference to r-value" compiler error and thus made it const - instead just:

`auto it = mapOfPcieDevicePaths.find(device)`

iterators are just pointers so tricks like this aren't saving any resources.

If you really wanted an iterator that can't be used to change the thing being pointed at you can't use auto because the container itself isn't const - instead you need something like:

`decltype(mapOfPcieDevicePaths)::const_iterator it =  mapOfPcieDevicePaths.find(device)`

but even that feels a bit unusual.